components/user-service/src/main/java/edu/jhuapl/dorset/fileuserservice/FileUserService.java

```diff
@@ -0,0 +1,139 @@
+ * limitations under the License.
+ */
+
+package edu.jhuapl.dorset.fileuserservice;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import edu.jhuapl.dorset.users.User;
+import edu.jhuapl.dorset.users.UserException;
+import edu.jhuapl.dorset.users.UserService;
+
+public class FileUserService implements UserService {
+    private static final Logger logger = LoggerFactory.getLogger(FileUserService.class);
+
+    private Map<String, User> users;
+    protected String userDirPath;
+    protected String fileBaseName;
+
+    /**
+     * 
+     * FileUserService
+     * 
+     * @param userDirectoryPath Path to directory that contains User files
+     * @param fileBaseName Base string of User file names
+     */
+    public FileUserService(String userDirectoryPath, String fileBaseName) {
+        users = new HashMap<String, User>();
+        this.userDirPath = userDirectoryPath;
+        this.fileBaseName = fileBaseName;
+    }
+
+    @Override
+    public String create(User user) throws UserException {
+        Properties prop = new Properties();
+        OutputStream output = null;
+        String userFilePath = this.userDirPath + this.fileBaseName + "-" + user.getUserName()
+                        + ".properties";
+
+        // Check if Username already exists
+        if (new File(userFilePath).exists()) {
+            throw new UserException(user.getUserName());
+        }
+
+        try {
+            output = new FileOutputStream(userFilePath);
+
+            Set<String> userKeys = user.getUserInformationKeys();
+            for (String key : userKeys) {
+                prop.setProperty(key, user.getUserInformation(key));
+            }
+            prop.store(output, null);
+
+        } catch (IOException e) {
+            throw new UserException("Username (" + user.getUserName() + ") already exists.");
+        }
+
+        this.users.put(user.getUserName(), user);
+        return user.getUserName();
+    }
+
+    @Override
+    public String retrieve(Properties properties) throws UserException {
+        User user = new User();
+        String userName = properties.getProperty("userName");
+
+        FileInputStream in;
+        Properties props = new Properties();
+        try {
+            in = new FileInputStream(this.userDirPath + "/" + this.fileBaseName + "-" + userName
+                            + ".properties");
+            props.load(in);
+            in.close();
+        } catch (IOException e) {
+            throw new UserException("User file (" + this.fileBaseName + "-" + userName
+                            + ".properties) does not exist.");
+        }
+
+        Enumeration<?> keySet = props.propertyNames();
```

Comment:
@cash props.keySet() cleans the code up some but there is casting that still needs to occur over the key/value pair

        Set<Object> keySet = props.keySet();
        for (Object key : keySet) {
            user.setUserInformation((String) key, props.getProperty((String) key));
        }